### PR TITLE
MSP-142: Updated trial stack to initialise new docker based AMI

### PR DESCRIPTION
**Pull request by gavincornwell**
NOTE: Still need to review the admin password process with Toni.

### CODE REVIEW

File: cloudformation/online-trial-stack.yaml [hunk 1]
```diff
@@ -38,10 +38,9 @@
     TrialInstanceType:
       Description: The type of EC2 instance to deploy the AMI to
       Type: String
-      Default: t2.medium
+      Default: t2.large
```
Comment (dcaruana) [inline cloudformation/online-trial-stack.yaml]: Does this mean we must now use t2.large?

Comment (gavincornwell) [reply cloudformation/online-trial-stack.yaml]: Yes, just pre-empting the result of MSP-147 and also means timings/testing are being done on the instance type we will likely be using.

**Comment by gavincornwell** [comment]
As discussed this morning, we will change the admin password CFN parameter to be the encoded value so we don't have to calculate it in UserData.

If that's not possible due to other dependencies we can look at creating a CFN custom resource to generate the md4 encoding and pass that to UserData.

**Comment by toniblyx** [comment]
Most of the userdata stuff can be moved to a script, including the admin password generator.

**Comment by gavincornwell** [comment]
That's true, but we'd still end up having to write the raw plain-text password into the script (which is better than userdata but still not great).

**Comment by toniblyx** [comment]
The script can be deleted afterwards, userdata info at the metadata server can not.

**Pull request merged**